src/modules/md/functions.cpp

```diff
@@ -32,3 +35,107 @@ double MDModule::determineTimeStep(const std::vector<Vec3<double>> &f)
 
     return 1.0 / fMax.absMax();
 }
+
+// Evolve Species coordinates, returning new coordinates
+std::vector<Vec3<double>> MDModule::evolve(const ProcessPool &procPool, const PotentialMap &potentialMap, const Species *sp,
+                                           double temperature, int nSteps, double deltaT,
+                                           const std::vector<Vec3<double>> &rInit, std::vector<Vec3<double>> &velocities)
+{
+    assert(sp);
+    assert(sp->nAtoms() == velocities.size());
+
+    // Create arrays
+    std::vector<double> mass(sp->nAtoms(), 0.0);
+    std::vector<Vec3<double>> forces(sp->nAtoms()), accelerations(sp->nAtoms());
+
+    // Variables
+    auto &atoms = sp->atoms();
+    double tInstant, ke, tScale;
+
+    // Units
+    // J = kg m2 s-2  -->   10 J = g Ang2 ps-2
+    // If ke is in units of [g mol-1 Angstroms2 ps-2] then must use kb in units of 10 J mol-1 K-1 (= 0.8314462)
+    const auto kb = 0.8314462;
+
+    // Store atomic masses for future use
+    for (auto &&[i, m] : zip(atoms, mass))
+        m = AtomicMass::mass(i.Z());
+
+    // Calculate total velocity and mass over all atoms
+    Vec3<double> vCom;
+    auto massSum = 0.0;
+    for (auto &&[v, m] : zip(velocities, mass))
+    {
+        vCom += v * m;
+        massSum += m;
+    }
```

Comment:
```suggestion
    auto massSum = std::accumulate(mass.begin(), mass.end());
    auto vCom = std::transform_reduce(mass.begin(), mass.end(), velocities.begin(), 0);
```

This should help the compiler with vectorization.